docs/api/sbm-model.rst

```diff
@@ -0,0 +1,220 @@
+
+.. autosummary::
+   :toctree: _generated
+
+   WflowSbmModel.setup_grid_from_raster
+   WflowSbmModel.setup_1dmodel_connection
+   WflowSbmModel.upgrade_to_v1_wflow
+   WflowSbmModel.clip
+
+
+Components
+----------
+
+If you are using python, you can access and update the model data using the components such as ``config``, ``staticmaps`` etc.
+This is for example useful for computing statictics, plotting etc.
+The components data are usually xarray, dictionnary or geopandas objects and can be accessed via the data property: `model.staticmaps.data`.
+The components of the WflowSbmModel are:
+
+
++--------------------------+-----------------------------------------------------------------+
+| **Model Component**      | **Component class**                                             |
++==========================+=================================================================+
+| ``model.config``         | :class:`~hydromt_wflow.components.WflowConfigComponent`         |
++--------------------------+-----------------------------------------------------------------+
+| ``model.staticmaps``     | :class:`~hydromt_wflow.components.WflowStaticmapsComponent`     |
++--------------------------+-----------------------------------------------------------------+
+| ``model.geoms``          | :class:`~hydromt_wflow.components.WflowGeomsComponent`          |
++--------------------------+-----------------------------------------------------------------+
+| ``model.forcing``        | :class:`~hydromt_wflow.components.WflowForcingComponent`        |
++--------------------------+-----------------------------------------------------------------+
+| ``model.states``         | :class:`~hydromt_wflow.components.WflowStatesComponent`         |
++--------------------------+-----------------------------------------------------------------+
+| ``model.tables``         | :class:`~hydromt_wflow.components.WflowTablesComponent`         |
++--------------------------+-----------------------------------------------------------------+
+| ``model.output_grid``    | :class:`~hydromt_wflow.components.WflowOutputGridComponent`     |
++--------------------------+-----------------------------------------------------------------+
+| ``model.output_scalar``  | :class:`~hydromt_wflow.components.WflowOutputScalarComponent`   |
++--------------------------+-----------------------------------------------------------------+
+| ``model.output_csv``     | :class:`~hydromt_wflow.components.WflowOutputCsvComponent`      |
++--------------------------+-----------------------------------------------------------------+
+
+I/O methods
+-------------
+
+If you are using python, you can read and write the different model components using the methods below.
+
+.. autosummary::
+   :toctree: _generated
+
+   WflowSbmModel.build
+   WflowSbmModel.update
+
+   WflowSbmModel.read
+   WflowSbmModel.write
```

Comment:
In the html you shared, I see that the component level methods are listed. I think this is good especially if users want to update specific options (especially for config and forcing). So I think good to list them again. Diff is that now they are WflowSbmModel.config.read instead of WflowSbmModel.read_config.

You could also list them in a separate table to make it clearer (ie for advanced options on individual components read and write, use: )

The other general methods that you have removed can stay removed